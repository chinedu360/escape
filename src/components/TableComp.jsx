import {
  Table,
  Thead,
  Tbody,
  // Tfoot,
  Tr,
  Th,
  Td,
  // TableCaption,
  TableContainer,
} from "@chakra-ui/react";

import { useNavigate } from "react-router-dom";
import { IoMdArrowBack } from "react-icons/io";

const TableComp = ({
  variant,
  colorScheme,
  title,
  display,
  height,
  scroll,
  overflow,
  titleName,
  show,
}) => {
  const name = "Chinedu"; //would come from an api
  const imageUrl = "";

  const navigate = useNavigate();

  const handleBack = () => {
    navigate(-1);
  };
  return (
    <div className={`m-2 rounded-lg bg-white shadow-lg`}>
      <h2 className="m-2 font-semibold text-[20px]">{title}</h2>
      <TableContainer borderRadius="lg">
        <Table variant={variant} colorScheme={colorScheme}>
          {/* <TableCaption>Imperial to metric conversion factors</TableCaption> */}
          <Thead>
            <Tr>
              <Th> {titleName}</Th>
              <Th display={display}>Description</Th>
              <Th display={display}>Date</Th>
              <Th display={display}>Writer</Th>
            </Tr>
          </Thead>
          <Tbody>
            <Tr>
              <Td className="flex items-center justify-start">
                <div className=" pr-2 flex-col items-start ">
                  <div className="flex">
                    {imageUrl || "" ? (
                      <img
                        src={imageUrl || ""}
                        alt="Profile"
                        className="rounded-full h-10 w-10"
                      />
                    ) : (
                      <div className="rounded-full h-10 w-10 bg-gray-300 flex justify-center items-center">
                        <span className="text-white font-bold text-sm">
                          {name.slice(0, 2).toUpperCase()}
                        </span>
                      </div>
                    )}
                  </div>
                </div>
                <p>inches</p>
              </Td>
              <Td display={display}>millimetres (mm)</Td>
              <Td display={display}>25.4</Td>
              <Td display={display}>Chinedu</Td>
            </Tr>
            <Tr>
              <Td className="flex items-center justify-start">
                <div className=" pr-2 flex-col items-start ">
                  <div className="flex">
                    {imageUrl || "" ? (
                      <img
                        src={imageUrl || ""}
                        alt="Profile"
                        className="rounded-full h-10 w-10"
                      />
                    ) : (
                      <div className="rounded-full h-10 w-10 bg-gray-300 flex justify-center items-center">
                        <span className="text-white font-bold text-sm">
                          {name.slice(0, 2).toUpperCase()}
                        </span>
                      </div>
                    )}
                  </div>
                </div>
                <p>inches</p>
              </Td>
              <Td display={display}>millimetres (mm)</Td>
              <Td display={display}>25.4</Td>
              <Td display={display}>Chinedu</Td>
            </Tr>
            <Tr>
              <Td className="flex items-center justify-start">
                <div className=" pr-2 flex-col items-start ">
                  <div className="flex">
                    {imageUrl || "" ? (
                      <img
                        src={imageUrl || ""}
                        alt="Profile"
                        className="rounded-full h-10 w-10"
                      />
                    ) : (
                      <div className="rounded-full h-10 w-10 bg-gray-300 flex justify-center items-center">
                        <span className="text-white font-bold text-sm">
                          {name.slice(0, 2).toUpperCase()}
                        </span>
                      </div>
                    )}
                  </div>
                </div>
                <p>inches</p>
              </Td>
              <Td display={display}>millimetres (mm)</Td>
              <Td display={display}>25.4</Td>
              <Td display={display}>Chinedu</Td>
            </Tr>
            <Tr>
              <Td className="flex items-center justify-start">
                <div className=" pr-2 flex-col items-start ">
                  <div className="flex">
                    {imageUrl || "" ? (
                      <img
                        src={imageUrl || ""}
                        alt="Profile"
                        className="rounded-full h-10 w-10"
                      />
                    ) : (
                      <div className="rounded-full h-10 w-10 bg-gray-300 flex justify-center items-center">
                        <span className="text-white font-bold text-sm">
                          {name.slice(0, 2).toUpperCase()}
                        </span>
                      </div>
                    )}
                  </div>
                </div>
                <p>inches</p>
              </Td>
              <Td display={display}>millimetres (mm)</Td>
              <Td display={display}>25.4</Td>
              <Td display={display}>Chinedu</Td>
            </Tr>
            <Tr>
              <Td className="flex items-center justify-start">
                <div className=" pr-2 flex-col items-start ">
                  <div className="flex">
                    {imageUrl || "" ? (
                      <img
                        src={imageUrl || ""}
                        alt="Profile"
                        className="rounded-full h-10 w-10"
                      />
                    ) : (
                      <div className="rounded-full h-10 w-10 bg-gray-300 flex justify-center items-center">
                        <span className="text-white font-bold text-sm">
                          {name.slice(0, 2).toUpperCase()}
                        </span>
                      </div>
                    )}
                  </div>
                </div>
                <p>inches</p>
              </Td>
              <Td display={display}>millimetres (mm)</Td>
              <Td display={display}>25.4</Td>
              <Td display={display}>Chinedu</Td>
            </Tr>
            <Tr>
              <Td className="flex items-center justify-start">
                <div className=" pr-2 flex-col items-start ">
                  <div className="flex">
                    {imageUrl || "" ? (
                      <img
                        src={imageUrl || ""}
                        alt="Profile"
                        className="rounded-full h-10 w-10"
                      />
                    ) : (
                      <div className="rounded-full h-10 w-10 bg-gray-300 flex justify-center items-center">
                        <span className="text-white font-bold text-sm">
                          {name.slice(0, 2).toUpperCase()}
                        </span>
                      </div>
                    )}
                  </div>
                </div>
                <p>inches</p>
              </Td>
              <Td display={display}>millimetres (mm)</Td>
              <Td display={display}>25.4</Td>
              <Td display={display}>Chinedu</Td>
            </Tr>
          </Tbody>
        </Table>
      </TableContainer>
      <div
        className={`${show} flex p-5 items-center cursor-pointer space-x-1`}
        onClick={handleBack}
      >
        <IoMdArrowBack /> <p>Back</p>
      </div>
    </div>
  );
};

export default TableComp;
