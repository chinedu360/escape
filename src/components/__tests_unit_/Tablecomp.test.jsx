import { render, screen } from "@testing-library/react";
import TableComp from "../TableComp";
import { MemoryRouter } from "react-router-dom"; // Use MemoryRouter for routing-related tests
import { vi, describe, it, expect } from "vitest";
import TestRenderer from "react-test-renderer";

describe("TableComp.jsx", () => {
  it("test the children inside the TableComp component", () => {
    const header = TestRenderer.create(
      <MemoryRouter>
        <TableComp />
      </MemoryRouter>
    ).toJSON();
    console.log(
      header.children[1].children[0].children[0].children[0].children[2]
    );
    //Header containers
    expect(header.type).toBe("div");
    expect(header.props.className).toBe(
      "m-2 rounded-lg bg-white shadow-lg h-70vh overflow-y-scroll"
    );
    expect(header.children[0].type).toBe("h2");
    expect(header.children[1].type).toBe("div");
    expect(header.children[1].children[0].type).toBe("table");
    expect(header.children[1].children[0].children[0].type).toBe("thead");
    expect(header.children[1].children[0].children[1].type).toBe("tbody");
    expect(header.children[1].children[0].children[0].children[0].type).toBe(
      "tr"
    );
    expect(
      header.children[1].children[0].children[0].children[0].children[0].type
    ).toBe("th");
    expect(
      header.children[1].children[0].children[0].children[0].children[2].type
    ).toBe("th");
    expect(
      header.children[1].children[0].children[0].children[0].children[2]
        .children[0]
    ).toBe("Description");
    expect(header.children[2].type).toBe("div");
    expect(header.children[2].props.className).toBe(
      "undefined flex p-5 items-center cursor-pointer space-x-1"
    );
  });

  it("Snapshot", () => {
    const header = TestRenderer.create(
      <MemoryRouter>
        <TableComp />
      </MemoryRouter>
    ).toJSON();

    expect(header).toMatchSnapshot();
  });
});
