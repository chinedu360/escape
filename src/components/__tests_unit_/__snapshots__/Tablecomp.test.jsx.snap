// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableComp.jsx > Snapshot 1`] = `
<div
  className="m-2 rounded-lg bg-white shadow-lg h-70vh overflow-y-scroll"
>
  <h2
    className="m-2 font-semibold text-20px"
    style={
      {
        "backgroundColor": "white",
        "position": "sticky",
        "top": 0,
        "zIndex": 1,
      }
    }
  >
     
  </h2>
  <div
    className="chakra-table__container css-1gc4pzd"
  >
    <table
      className="chakra-table css-0"
    >
      <thead
        className="css-0"
      >
        <tr
          className="css-0"
        >
          <th
            className="css-0"
          >
             
          </th>
           
          <th
            className="css-0"
          >
            Description
          </th>
        </tr>
      </thead>
      <tbody
        className="css-0"
      />
    </table>
  </div>
  <div
    className="undefined flex p-5 items-center cursor-pointer space-x-1"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M427 234.625H167.296l119.702-119.702L256 85 85 256l171 171 29.922-29.924-118.626-119.701H427v-42.75z"
      />
    </svg>
     
    <p>
      Back
    </p>
     
  </div>
</div>
`;
